core/kt-osrd-sim-infra/src/main/kotlin/fr/sncf/osrd/sim_infra/impl/RawInfraBuilder.kt

```diff
@@ -140,38 +157,74 @@ class RouteDescriptorImpl(
     override val speedLimits: StaticIdxList<SpeedLimit>,
     override val speedLimitStarts: DistanceList,
     override val speedLimitEnds: DistanceList,
+    override val chunks: DirStaticIdxList<TrackChunk>,
 ) : RouteDescriptor
 
 interface RestrictedRawInfraBuilder {
-    fun movableElement(delay: Duration, init: MovableElementDescriptorBuilder.() -> Unit): MovableElementId
+    fun movableElement(delay: Duration, init: MovableElementDescriptorBuilder.() -> Unit): TrackNodeId
     fun detector(name: String?): DetectorId
     fun linkZones(zoneA: ZoneId, zoneB: ZoneId): DetectorId
     fun linkZones(detector: DetectorId, zoneA: ZoneId, zoneB: ZoneId)
     fun setNextZone(detector: DirDetectorId, zone: ZoneId)
-    fun zone(movableElements: StaticIdxSortedSet<MovableElement>): ZoneId
-    fun zone(movableElements: List<MovableElementId>): ZoneId
-    fun zone(movableElements: StaticIdxSortedSet<MovableElement>, bounds: List<DirDetectorId>): ZoneId
+    fun zone(movableElements: StaticIdxSortedSet<TrackNode>): ZoneId
+    fun zone(movableElements: List<TrackNodeId>): ZoneId
+    fun zone(movableElements: StaticIdxSortedSet<TrackNode>, bounds: List<DirDetectorId>): ZoneId
```

Comment:
Don't we need explicit zones for the signaling simulation? I'm not quite sure what the implications would be here.

Reply:
when this API was first designed, we could not just generate zones from track topology using a union find. As your PR adds track topology, we can now do just that, which alleviates the need for manually setting zones up. It can be automatically done.

Reply:
For now we just "import" the zones from the legacy infra. We can rewrite the generation using the topology but I don't think it's the right time, I'd group this with the general rewrite of rjs parsing and the eventual removal of the legacy infra. 

I could add a comment there saying that these 3 methods should be removed from the interface at that moment.